### PR TITLE
feat: add AI Agent Guidelines for SAP UX Tools

### PR DESCRIPTION
**Pull request by devinea**

### CODE REVIEW

File: AGENTS.md [hunk 1]
```diff
@@ -0,0 +1,602 @@
...
+- [Development Standards](#development-standards)
+- [Package Management](#package-management)
+- [Testing Requirements](#testing-requirements)
+- [Security and Maintenance](#security-and-maintenance)
+- [Change Management](#change-management)
+- [Useful Commands](#useful-commands)
+
+## Overview
+
+The SAP UX Tools is a monorepo containing 96+ packages for building SAP Fiori applications. The project uses:
```
Comment (Klaus-Keller) [inline AGENTS.md]: Should we really name the number? Wouldn't it be better to state something like all packages of this monorepo can be found in /packages folder

File: AGENTS.md [hunk 2]
```diff
@@ -0,0 +1,602 @@
...
+
+# Check specific package
+pnpm outdated --filter @sap-ux/[package-name]
+```
+
+### 3. Code Coverage Standards
+
+**Minimum code coverage: 80% for all code changes and packages**
+
+**Requirements:**
```
Comment (Klaus-Keller) [inline AGENTS.md:87]: Should we mention that test files are having the naming pattern `*.test.ts`, usually with the name of the file in `src` they test?

File: AGENTS.md [hunk 3]
```diff
@@ -0,0 +1,602 @@
...
+
+**Standards:**
+- TypeScript 5.9.3+ (current version in repo)
+- Target: ES2023
+- Strict mode enabled
+- Use ES modules (import/export)
+- Prefer `const` over `let`
+- Use async/await over raw Promises
+- Leverage TypeScript features: generics, union types, type guards, etc.
+- Avoid `any` type - use `unknown` or proper types
```
Comment (Klaus-Keller) [inline AGENTS.md:163]: do we want to avoid `enum`s? If yes we could write it here

File: AGENTS.md [hunk 4]
```diff
@@ -0,0 +1,602 @@
...
+### Code Style
+
+Follow the conventions in [docs/Guidelines.md](docs/Guidelines.md):
+
+**TypeScript:**
+- Use ESLint and Prettier for consistent formatting
+- 120 character line length
+- 4 space indentation
+- Single quotes
+- No trailing commas
```
Comment (Klaus-Keller) [inline AGENTS.md]: we have the same info already above

File: AGENTS.md [hunk 5]
```diff
@@ -0,0 +1,602 @@
...
+# With coverage
+pnpm test # Coverage is collected by default
+```
+
+**Best practices:**
+- Use `describe` blocks to group related tests
+- Use clear, descriptive test names
+- Follow given/when/then structure
+- Test edge cases and error conditions
+- Use snapshots for generated files
```
Comment (Klaus-Keller) [inline AGENTS.md:431]: we could add a sentence to make sure all tests run on Linux, Mac, and Windows, e.g. snapshots should never contain file or folder paths with `/` or `\`

File: AGENTS.md [hunk 6]
```diff
@@ -0,0 +1,602 @@
...
+- [Testing Requirements](#testing-requirements)
+- [Security and Maintenance](#security-and-maintenance)
+- [Change Management](#change-management)
+- [Useful Commands](#useful-commands)
+
+## Overview
+
+The SAP UX Tools is a monorepo containing 96+ packages for building SAP Fiori applications. The project uses:
+
+- **Package Manager**: pnpm 8.14.0 (required)
```
Comment (heimwege) [inline AGENTS.md]: soon pnpm 10 😄 Should we really include such detailed version information? That will be outdated sooner or later and in case we forget to adjust...

File: AGENTS.md (hunk 3 above)
Comment (heimwege) [inline AGENTS.md:163]: Avoid `any` 😍

File: AGENTS.md [hunk 7]
```diff
@@ -0,0 +1,602 @@
...
+   - Package folders: `packages/[name]`
+   - Folder name = package name without `@sap-ux/` prefix
+   - Example: `@sap-ux/fiori-freestyle-writer` → `packages/fiori-freestyle-writer`
+
+3. **Extension naming:**
+   - Pattern: `sap-ux-[name]-ext`
+   - Examples: `sap-ux-application-modeler-ext`, `sap-ux-sap-systems-ext`
+
+4. **Private packages:**
+   - Must have `"private": true` in package.json
```
Comment (heimwege) [inline AGENTS.md]: There'S no help on when to decide if a package is private. Wouldn't this lead to new packages being private by mistake?

File: AGENTS.md [hunk 8]
```diff
@@ -0,0 +1,602 @@
...
+
+**All changes to source code or runtime dependencies require a changeset.**
+
+**When to create a changeset:**
+- ✅ Any changes to files in `src/` directory
+- ✅ Adding, removing, or updating dependencies (not devDependencies)
+- ✅ Changes to templates or runtime assets
+- ✅ Bug fixes
+- ✅ New features
+- ✅ Breaking changes
```
Comment (heimwege) [inline AGENTS.md:205]: changes to README.md ?

File: AGENTS.md [hunk 9]
```diff
@@ -0,0 +1,602 @@
...
+
+**When NOT to create a changeset:**
+- ❌ Changes only to tests
+- ❌ Changes only to devDependencies and not using esbuild for bundling
+- ❌ Documentation updates
```
Comment (heimwege) [inline AGENTS.md]: Not sure about that.

File: AGENTS.md [hunk 10]
```diff
@@ -0,0 +1,602 @@
...
+
+# View coverage reports
+# Coverage reports are in: packages/[package-name]/coverage/lcov-report/index.html
+```
+
+**Coverage Configuration** (from [jest.base.js](jest.base.js:6)):
+- `collectCoverageFrom: ['src/**/*.ts']`
+- Reports: text and lcov formats
+- SonarCloud integration via jest-sonar reporter
+
```
Comment (heimwege) [inline AGENTS.md:106]: I'm missing the integration tests here

File: AGENTS.md (hunk 4 above)
Comment (heimwege) [reply AGENTS.md]: can we maybe just link to the rules we are using? Maybe the eslint rules file of the respective package or something?

File: AGENTS.md [hunk 11]
```diff
@@ -0,0 +1,602 @@
...
+4. **Don't resolve conversations** - Let reviewers mark as resolved
+5. **AI-generated code** - Follow [SAP's GenAI guidelines](https://github.com/SAP/.github/blob/main/CONTRIBUTING_USING_GENAI.md)
+
+## Package Management
+
+### Workspace Structure
+
+**Root**: [pnpm-workspace.yaml](pnpm-workspace.yaml)
+```yaml
+packages:
```
Comment (heimwege) [inline AGENTS.md:357]: `packages:` and the having a folder `packages/**` confuses me. it AI fine with that?

File: AGENTS.md [hunk 12]
```diff
@@ -0,0 +1,602 @@
...
+
+```
+packages/[package-name]/
+├── src/              # Source code (TypeScript)
+├── test/             # Unit tests
+├── templates/        # Template files (for generators)
+├── dist/             # Build output (generated)
+├── coverage/         # Test coverage (generated)
+├── package.json      # Package manifest
+├── tsconfig.json     # TypeScript config
```
Comment (heimwege) [inline AGENTS.md:375]: eslint config missing

File: AGENTS.md [hunk 13]
```diff
@@ -0,0 +1,602 @@
...
+├── jest.config.js    # Jest config
+└── README.md         # Package documentation
+```
+
+### Workspace Dependencies
+
+Use `workspace:*` for internal dependencies:
+
+```json
```
Comment (heimwege) [inline AGENTS.md:386]: This is not marked as Example. Might be misleading for AI to only be a rule to these two packages 🤔

**Comment by devinea** [comment]
@Klaus-Keller  Thanks for reviewing. 
I've addressed the feedback. https://github.com/SAP/open-ux-tools/pull/4134/commits/0e8907e4a09e347dde4e6176da54865f18d40057

File: AGENTS.md [hunk 14]
```diff
@@ -0,0 +1,602 @@
...
+
+# Check dependency versions
+pnpm lint:dependency-versions
+```
+
+### Testing
+
+```bash
+# Run all tests
+pnpm test
```
Comment (heimwege) [inline AGENTS.md:569]: do we need to add that when working on package A only tests for that package should be executed? I alway yell at AI in case it wants to run pnpm test in the root 🙈

File: AGENTS.md (hunk 10 above)
Comment (devinea) [reply AGENTS.md:106]: I don't think we merge coverage from integration tests with unit tests  in this repo. 
https://github.com/SAP/open-ux-tools/blob/main/jest.base.js#L24

File: AGENTS.md (hunk 11 above)
Comment (devinea) [reply AGENTS.md:357]: That is the content from `pnpm-workspace.yaml` so not much to say on that.

**Comment by devinea** [comment]
@heimwege  Thanks for reviewing.
I've addressed the feedback. https://github.com/SAP/open-ux-tools/pull/4134/commits/464772637dfe74303b38366aa2246666ad751eda
